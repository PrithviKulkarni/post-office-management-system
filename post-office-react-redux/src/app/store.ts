import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import letterReducer from '../features/LetterSlice';
import parcelReducer from '../features/ParcelSlice';
import depotReducer from '../features/DepotSlice';
import driverReducer from '../features/DriverSlice' 
export const store = configureStore({
  reducer: {
    letter: letterReducer,
    parcel: parcelReducer,
    depot: depotReducer,
    driver: driverReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
