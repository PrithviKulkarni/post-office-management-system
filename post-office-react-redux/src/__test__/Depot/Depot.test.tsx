import { render, cleanup } from '@testing-library/react';
import AllDepots from '../../components/Depot/Depots';
import { screen } from '@testing-library/react';
import { Provider } from "react-redux";
import { store } from '../../app/store';

afterEach(cleanup);
it("Depots render test", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <AllDepots />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Verifies depots has rendered", () => {
  render(
    <Provider store={store}>
      <AllDepots />
    </Provider>
  );
  
  expect(screen.getByTestId("depotsPage")).toBeInTheDocument();
});