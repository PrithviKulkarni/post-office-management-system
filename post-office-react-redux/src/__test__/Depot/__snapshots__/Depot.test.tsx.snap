// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Depots render test 1`] = `
<DocumentFragment>
  <div
    data-testid="depotsPage"
  >
    <div
      class="accordion"
    />
  </div>
</DocumentFragment>
`;
